object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -765,13 +761,19 @@ private void propagate(
   private void checkMustCall(
       LocalVarWithAssignTree assign, CFStore store, String outOfScopeReason) {
     List<String> mustCallValue = getMustCallValue(assign.assignTree);
+    // optimization: if there are no must-call methods, we do not need to perform the check
+    if (mustCallValue.isEmpty()) {
+      return;
+    }
     AnnotationMirror dummyCMAnno = createCalledMethods(mustCallValue.toArray(new String[0]));
 
     boolean report = true;
 
     AnnotationMirror cmAnno;
 
-    CFValue lhsCFValue = store.getValue(assign.localVar);
+    // sometimes the store is null!  this looks like a bug in checker dataflow.
```

Comment:
@msridhar When I revert this hunk, the tests still pass.  It seems that the test case `MustCallNullStore.java` does not reproduce the bug.

Reply:
@mernst you may need to also comment out the emptiness check under the `// optimization` comment.  That may be masking the issue.  Sorry, I should have documented better how to reproduce.

Reply:
@msridhar I cannot reproduce the problem in that way, or by going back to a previous commit and adding the `MustCallNullStore.java` file.

Reply:
Hi Mike, if I do `git checkout 39774772fbbb4c2c0382425549fbbf6f51c350ed` (which in my history is the commit before this got merged to `always-call-before-occ`), re-add the `MustCallNullStore.java` file, and then run `./gradlew :object-construction-checker:test`, I see the failure and the NPE:

```
1 unexpected diagnostic was found:
:-1: other: error: SourceChecker.typeProcess: unexpected Throwable (NullPointerException) while processing tests/mustcall/MustCallNullStore.java
  ; The Checker Framework crashed.  Please report the crash.
  Compilation unit: tests/mustcall/MustCallNullStore.java
  Last visited tree at line 5 column 1:
  class MustCallNullStore {
  Exception: java.lang.NullPointerException; java.lang.NullPointerException
  	at org.checkerframework.checker.objectconstruction.ObjectConstructionAnnotatedTypeFactory.checkMustCall(ObjectConstructionAnnotatedTypeFactory.java:774)
  	at org.checkerframework.checker.objectconstruction.ObjectConstructionAnnotatedTypeFactory.mustCallTraverse(ObjectConstructionAnnotatedTypeFactory.java:544)
  	at org.checkerframework.checker.objectconstruction.ObjectConstructionAnnotatedTypeFactory.postAnalyze(ObjectConstructionAnnotatedTypeFactory.java:342)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.analyze(GenericAnnotatedTypeFactory.java:1358)
```

Does that work for you?

Reply:
That is exactly what I did.  Both `./gradlew test` and `./gradlew :object-construction-checker:test` work for me.  We must be doing something differently.  Could you create a new branch with your state and push it, so we can see the CI output?

Reply:
Here's the branch:

https://github.com/msridhar/object-construction-checker/tree/test-null-store-failure

Here's the failing build:

https://travis-ci.com/github/msridhar/object-construction-checker/builds/197925335

Perhaps this is an issue with using JDK 8 vs JDK 11?

Reply:
I needed to run `./gradlew clean` before `./gradlew test`; otherwise `./gradlew test` passed.  This seems like a bug in the build system.  Can someone who knows Gradle better than I do look into that?

Reply:
I think this is the right line to ensure tests are re-run when a new test input is added:

https://github.com/kelloggm/object-construction-checker/blob/d94c75550da693b9b1cd02ec20e73c3984cbdcd8/object-construction-checker/build.gradle#L86

I just tested locally and when I add a new Java file under the `object-construction-checker/tests/mustcall` directory the tests get re-run even without cleaning.  So I am not sure why you need to run clean first 🤔